### ISSUE TITLE
IDE crashes attempting to resolve [ConstEval]

### ISSUE THREAD

**Issue by gopakumar-gg** [beefytech/Beef#1694]
IDE crashes upon loading below snippet.  (Tested with  nightly builds 24th & 25th Aug)

```csharp
		[Comptime(ConstEval = true)]
		static int Fun<T>(T num) where T : const int
		{
			return [ConstEval]Fun(num);  // CRASH HERE.   Is it calling recursively? 

			int val = num;
			// return Fun(val); // This works fine and does the job, as expected
		}

		static int Fun(int num)
		{
			return num * num;
		}
		
		public static void TestCTFE()
		{
			int a1 = Fun(1);
			int a2 = Fun(2);
			int x = 3;
			int a3 = Fun(x);
		}
```

**Comment by bfiete** [comment]
Fixed at 9062662ffb5960cf31a21f75e1a5696fcd7cc2a3

**Comment by gopakumar-gg** [comment]
If you build this code, IDE crashes, even after the above the fix.  I tested with Aug 26th nightly build.

**Comment by bfiete** [comment]
You posted this _very_ close to the time when the Aug 26th build actually went live - are you 100% you actually used the Aug 26th build?

I just downloaded it myself and in my test this code _does not_ crash, and I verified it does crash on the Aug 26th build still.

Can you post your crash dump?

**Comment by gopakumar-gg** [comment]
I was waiting for the Aug 26 build to come out to test this, after seeing that this issue got closed.   I posted immediately. 
Two crash dumps attached.
[CrashDumps.zip](https://github.com/beefytech/Beef/files/9438475/CrashDumps.zip)
.

**Comment by bfiete** [comment]
Got it- that part is fixed at 0b162c0ebde8221cd206c138137cc5a2579d94b6
